CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -198,11 +198,47 @@ def __init__(self, db_client: 'botocore.client.DynamoDB'):
         super().__init__(db_client=db_client)
 
 class FridgeHistory(DB_Item):
-    
+
     def __init__(self, db_client: 'botocore.client.DynamoDB'):
         super().__init__(db_client=db_client)
 
-class Tag(DB_Item):
 
-    def __init__(self, db_client: 'botocore.client.DynamoDB'):
+class Tag(DB_Item):
+    TABLE_NAME = "tag"
+    def __init__(self, db_client: 'botocore.client.DynamoDB', tag_name:str= None):
         super().__init__(db_client=db_client)
+
+        self.db_client = db_client
+        self.tag_name = tag_name
+
+    def set_tag(self, tag_name):
+        #Tag_name is alphanumeric, can include hyphen and underscore, with no spaces and all lower cased
+        tag_name = tag_name.lower().replace(" ", "")
+        if not self.is_valid_tag_name(tag_name):
+            return False
+        else:
+            self.tag_name = tag_name
+            return True
+
+    def is_valid_tag_name(self, tag_name) --> bool:
+        #a valid tag name is alphanumeric, can include hyphen, underscore
+        for x in tag_name:
+            if not x.isalnum() and x not in ['_', '-']:
+                return False
+        return True
+
+    def add_item(self) -> DB_Response:
+        is_valid_field = self.is_valid_tag_name(self.tag_name)
+        if not is_valid_field:
+            return DB_Response(message = "Tag Name Can Only Contain Letters, Numbers, Hyphens and Underscore: %s" % self.tag_name, status_code=400, success=False)
```

Comment:
style - this line is a bit long, can we make it multi-line?

DB_Resonse(
    message="",
    status_code=...
)